frappe/integrations/doctype/ldap_settings/ldap_settings.py

```diff
@@ -142,6 +148,61 @@ def get_ldap_attributes(self):
 
 		return ldap_attributes
 
+
+	def fetch_ldap_groups(self, user, conn):
+		fetch_ldap_groups = None
+
+		ldap_attributes = ['cn']
+
+		ldap_object_class = None
+		ldap_group_members_attribute = None
+		ldap_group_search_filter = None
+
+
+		if self.ldap_directory_server.lower() == 'active directory':
+
+			ldap_object_class = 'Group'
+			ldap_group_members_attribute = 'member'
+			user_search_str = user.entry_dn
+
+
+		elif self.ldap_directory_server.lower() == 'openldap':
+
+			ldap_object_class = 'posixgroup'
+			ldap_group_members_attribute = 'memberuid'
+			user_search_str = getattr(user, self.ldap_username_field).value
+
+		elif self.ldap_directory_server.lower() == 'custom':
+
+			ldap_object_class = self.ldap_group_objectclass
+			ldap_group_members_attribute = self.ldap_group_member_attribute
+			user_search_str = frappe.as_unicode(frappe.form_dict.usr)
+
+		else:
+			# NOTE: depreciate this else path
+			# this path will be hit for everyone with preconfigured ldap settings. this must be taken into account so as not to break ldap for those users.
+
+			if self.ldap_group_field:
+
+				fetch_ldap_groups = getattr(user, self.ldap_group_field).values
+
+		if ldap_object_class is not None:
+			conn.search(
```

Comment:
Possible future bug: 

The user can edit the `organizational_unit_for_groups` field. Place this call within an try except block to capture user errors. The error should just report to the debug interface as only an administrator can fix. No website user dialog requured

Reply:
adjusted implementation.

implemented in 1c74ce19d5b613e306580e92cacaa8713d0cab66

conduct the validation of the search paths at the time of input.